blockchain/types/tx_internal_data_fee_delegated_smart_contract_execution.go

```diff
@@ -347,8 +347,8 @@ func (t *TxInternalDataFeeDelegatedSmartContractExecution) Validate(stateDB Stat
 }
 
 func (t *TxInternalDataFeeDelegatedSmartContractExecution) ValidateMutableValue(stateDB StateDB, currentBlockNumber uint64) error {
-	if !validate7702(stateDB, t.Type(), t.From, t.Recipient) {
-		return kerrors.ErrNotProgramAccount
+	if err := validate7702(stateDB, t.Type(), t.From, t.Recipient); err != nil {
```

Comment:
I'm not sure if it'll lead to problem, but it will result in different error returns (from `ErrNotProgramAccount` to `ErrFromMustBeEOAWithoutCode`. Do you think it's ok?

Reply:
Of course, it doesn't lead to state changes for historical block.

Reply:
Actually L350 is added in v2.0. So it shouldn't affect previous transactions.
```
[v1.0]
!IsContractAvailable -> ErrNotProgramAccount
```
```
[v2.0]
// before SetCodeTx, all transactions pass this condition, because tx.From is obviously an EOA and there is no EOA with code.
validate7702 -> nil or ErrFromMustBeEOAWithoutCode
// same condition kept.
!IsContractAvailable -> ErrNotProgramAccount
```

Reply:
As I mentioned in the description, I'm not sure what to do about this either.
The error returned for EOA without Code has been changed.
https://github.com/kaiachain/kaia/pull/182/files#diff-36960dea371fb7a7722f48b0905c890030f61801a29c132b5ade154357f0e1dbR785

Reply:
Because there may be external tests that expects ErrNotProgramAccount, so it might be better to change it back to it.

Reply:
Oh I misevaluated. The error message does change when `to` is an EOA without code.

However, "not a program account" is original in Kaia, so it shouldn't break Ethereum compatibility.
Also having clearer error message is not a bad idea I think.

Reply:
Ok, let's return a more accurate message. Thank you.